### PR TITLE
Disabling unused monitoring services

### PR DESCRIPTION
**Pull request by cappetersson**
PR connected to issue: https://github.com/openshift/hypershift/issues/2679

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -3441,11 +3442,13 @@ func (r *HostedClusterReconciler) reconcileMachineConfigServer(ctx context.Conte
 	}
 
 	// Reconcile service
-	mcsService := ignitionserver.MCSService(controlPlaneNamespace.Name)
-	if _, err := createOrUpdate(ctx, r.Client, mcsService, func() error {
-		return reconcileMachineConfigServerService(mcsService)
-	}); err != nil {
-		return fmt.Errorf("failed to reconcile machine config server service: %w", err)
+	if _, exists := hcluster.Annotations[hyperv1.DisableMonitoringServices]; !exists {
```
Comment (csrwng) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:3445]: can you please just remove this service? This is not used anymore.
cc @enxebre

**Review by csrwng — COMMENTED** [review]
Just one comment, otherwise lgtm

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go (hunk 1 above)
Comment (cappetersson) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:3445]: Thank you for the review Cesar. Just to make sure we are on the same page, are you referring to removing the whole `if _, exists := hcluster.Annotations[hyperv1.DisableMonitoringServices]; !exists {` if-statement section?

Comment (csrwng) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:3445]: yes

**Review by hasueki — APPROVED** [review]
/lgtm

**Pull request merged**